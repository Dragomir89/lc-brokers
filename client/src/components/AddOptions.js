import React, { Component } from 'react';
import TextInput from './common/TextInput';

class AddOpitonsPage extends Component {

    state = {
        constructionType: '',
        estateType: '',
        state: '',
        neighborhood: ''
    }

    changeFn = (e) => {
        const { name, value } = e.target;
        this.setState(() => {
            return { [name]: value }
        });
    }

    handleClick = (e) => {
        e.preventDefault();
        console.log(this.state);
        this.setState(() => {
            return  {
                constructionType: '',
                estateType: '',
                state: '',
                neighborhood: ''
            } 
        });
    }

    render () {
        return (
            <div className='content-container'>
                <h1 className='title__page'>Добвавете опции за офертите</h1>
                <form>
                    <TextInput
                        value={this.state.constructionType}
                        label='Вид Стоителство'
                        name='constructionType'
                        placeholder='старо, ново, ЕПК ...' changeFn={this.changeFn} />
                    <TextInput
                        value={this.state.estateType}
                        label='Вид Имот'
                        name='estateType'
                        placeholder='едностаен, двустаен, къща ...'
                        changeFn={this.changeFn} />
                    <TextInput
                        value={this.state.state}
                        label='Състояние'
                        name='state'
                        placeholder='нает, свободен ...'
                        changeFn={this.changeFn} />
                    <TextInput
                        value={this.state.neighborhood}
                        label='Квартал'
                        name='neighborhood'
                        placeholder='лозенец, цетър, стрелбище ...'
                        changeFn={this.changeFn} />
                    <button 
                        onClick={this.handleClick}
                        className='button__primary'>
                        Запази
                    </button>
                </form>
            </div>
        )
    }
}

export default AddOpitonsPage;