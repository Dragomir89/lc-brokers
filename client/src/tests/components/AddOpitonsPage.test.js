import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import '../setupTests';
import AddOptions from '../../components/AddOptions';

let wrapper;

beforeEach(() => {
    wrapper = shallow(<AddOptions />);
});

test('should render AddOptions component correctly', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
});

test('TextInput should call changeFn and set new state', () => {
    const event = { target: { value: 'test val', name: 'constructionType' } };
    wrapper.find('TextInput').at(0).prop('changeFn')(event);
    expect(wrapper.state([event.target.name])).toBe(event.target.value);
});

test('should on click save button to set default state', () => {
    const event = { preventDefault: () => { } };
    wrapper.find('button').simulate('click', event);
    expect(wrapper.state()).toEqual({
        constructionType: '',
        estateType: '',
        state: '',
        neighborhood: ''
    });
});