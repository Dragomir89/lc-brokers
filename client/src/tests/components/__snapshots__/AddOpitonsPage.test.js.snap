// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddOptions component correctly 1`] = `
<div
  className="content-container"
>
  <h1
    className="title__page"
  >
    Добвавете опции за офертите
  </h1>
  <form>
    <TextInput
      changeFn={[Function]}
      label="Вид Стоителство"
      name="constructionType"
      placeholder="старо, ново, ЕПК ..."
      value=""
    />
    <TextInput
      changeFn={[Function]}
      label="Вид Имот"
      name="estateType"
      placeholder="едностаен, двустаен, къща ..."
      value=""
    />
    <TextInput
      changeFn={[Function]}
      label="Състояние"
      name="state"
      placeholder="нает, свободен ..."
      value=""
    />
    <TextInput
      changeFn={[Function]}
      label="Квартал"
      name="neighborhood"
      placeholder="лозенец, цетър, стрелбище ..."
      value=""
    />
    <button
      className="button__primary"
      onClick={[Function]}
    >
      Запази
    </button>
  </form>
</div>
`;
